Source/SwiftLintCore/Models/Baseline.swift

```diff
@@ -0,0 +1,186 @@
+        )
+        let baselineViolationsByRuleIdentifier = baselineViolations.groupedByRuleIdentifier(
+            filteredBy: relativePathViolations
+        )
+
+        var filteredViolations: Set<BaselineViolation> = []
+
+        for (ruleIdentifier, ruleViolations) in violationsByRuleIdentifier {
+            guard
+                let baselineViolations = baselineViolationsByRuleIdentifier[ruleIdentifier],
+                    baselineViolations.isNotEmpty else {
+                filteredViolations.formUnion(ruleViolations)
+                continue
+            }
+
+            let groupedRuleViolations = Dictionary(grouping: ruleViolations, by: \.key)
+            let groupedBaselineViolations = Dictionary(grouping: baselineViolations, by: \.key)
+
+            for (key, ruleViolations) in groupedRuleViolations {
+                guard let baselineViolations = groupedBaselineViolations[key] else {
+                    filteredViolations.formUnion(ruleViolations)
+                    continue
+                }
+                if ruleViolations.count > baselineViolations.count {
+                    filteredViolations.formUnion(ruleViolations)
+                }
+            }
+        }
+
+        let violationsWithAbsolutePaths = Set(filteredViolations.violationsWithAbsolutePaths)
+        return violations.filter { violationsWithAbsolutePaths.contains($0) }
+    }
+
+    /// Returns the violations that are present in another `Baseline`, but not in this one.
+    ///
+    /// The violations are filtered using the same algorithm as the `filter` method above.
+    ///
+    /// - parameter otherBaseline: The other `Baseline`.
+    public func compare(_ otherBaseline: Baseline) -> [StyleViolation] {
+        otherBaseline.baselineViolations.flatMap {
+            filter($1.violationsWithAbsolutePaths)
+        }
+    }
+}
+
+private extension Sequence where Element == StyleViolation {
+    var baselineViolations: [BaselineViolation] {
+        var lines: [String: [String]] = [:]
+        var baselineViolations: [BaselineViolation] = []
+        for violation in self {
+            guard let absolutePath = violation.location.file,
+                  let lineNumber = violation.location.line != nil ? (violation.location.line ?? 0) - 1 : nil,
+                  lineNumber > 0 else {
+                baselineViolations.append(violation.baselineViolation())
+                continue
+            }
+            if let fileLines = lines[absolutePath] {
+                let text = (!fileLines.isEmpty && lineNumber < fileLines.count ) ? fileLines[lineNumber] : ""
+                baselineViolations.append(violation.baselineViolation(text: text))
+                continue
+            }
+            let text: String
+            if let fileLines = SwiftLintFile(path: absolutePath)?.lines.map(\.content),
+               lineNumber < fileLines.count {
+                text = fileLines[lineNumber]
+                lines[absolutePath] = fileLines
```

Comment:
Shouldn't the content be cached even if `lineNumber < fileLines.count` is not fulfilled?